libs/blog/newsletter/data-access/src/lib/state/newsletter.store.ts

```diff
@@ -0,0 +1,64 @@
+import { computed, inject } from '@angular/core';
+import { tapResponse } from '@ngrx/operators';
+import {
+  patchState,
+  signalStore,
+  withComputed,
+  withHooks,
+  withMethods,
+  withState,
+} from '@ngrx/signals';
+import { rxMethod } from '@ngrx/signals/rxjs-interop';
+import { pipe, switchMap, tap } from 'rxjs';
+
+import { SubscriberModel } from '../dto';
+import { NewsletterService } from '../infrastructure/newsletter.service';
+
+type NewsletterState = {
+  loading: 'success' | 'error' | 'init' | 'loading';
+};
+
+const initialState: NewsletterState = {
+  loading: 'init',
+};
+
+export const NewsletterStore = signalStore(
+  { providedIn: 'root' },
+  withState(initialState),
+  withComputed(({ loading }) => ({
+    loading: computed(() => loading()),
+  })),
+  withMethods((store, newsletterService = inject(NewsletterService)) => ({
+    resetToInit: () => {
+      patchState(store, { loading: 'init' });
+    },
+    postEmailAddress: rxMethod<SubscriberModel>(
+      pipe(
+        tap(() => {
+          patchState(store, {
+            loading: 'loading',
+          });
+        }),
+        switchMap((subscriber) =>
+          newsletterService.postEmailAddress(subscriber).pipe(
+            tapResponse({
+              next: () => {
+                patchState(store, { loading: 'success' });
+              },
+              error: () => {
+                patchState(store, { loading: 'error' });
+              },
+            }),
+          ),
+        ),
+      ),
+    ),
+  })),
+  withHooks((store) => {
+    return {
+      onDestroy() {
+        patchState(store, { loading: 'init' });
+      },
```

Comment:
if it's destroyed what's the point of patching the store?

Reply:
That's right, removed